### PR TITLE
Merge 'Cross-section analysis' module with 'Centerline metrics'

### PR DESCRIPTION
**Pull request by chir-set**
Hi @lassoan,

Following #36

'Cross-section analysis' core functionalities are hereby added to 'Centerline metrics'.

Namely :

 - orthogonal reslicing to centerline points
 - report slice view orientation in Slicer's coordinate system
 - distance of point from a relative origin.
 
Please consider this patch for merge in 'Centerline metrics'.

Regards.

### CODE REVIEW

**Comment by chir-set** [comment]
One functionality of 'Cross-section analysis' is skipped here.

It concerns orthogonal reslicing along arbitrary markups curves, that do not have radii scalars and are freely drawn.

Adding this functionality to 'Centerline metrics' as it is now would be messy and confusing, both in the UI and in code. Moreover, it's out of scope of a module dealing with centerlines.

I propose that 'Cross-section analysis' be removed from the VMTK group, and moved verbatim to 'Legacy' or 'Utilities'.

What's your take on this ?

**Comment by lassoan** [comment]
Thanks a lot for working on this! 

We already reslice image the image along the curve. We just need to offer the users to choose between reslice modes:
- position only: this is the current behavior
- minimum torsion: this is what the curve-to-world coordinate system provides (uses parallel transport), it can take an additional rotation slider as input to spin the cross-section to an anatomically meaningful orientation
- snap to anatomical axis: automatically spins the slice from the orientation that it got from minimum torsion so that up view up is superior

After this is implemented we can simply remove the `Cross-section analysis` module from the package and we don't have to maintain it anymore.

**Comment by chir-set** [comment]
I'm afraid what you envision is out of reach to me. I just don't have a clue about things like 'direction/normal/binormal vectors' or 'spin the cross-section to an anatomically meaningful orientation'.

Doesn't the reformat widget allow to spin a slice view to any orientation already ?

**Comment by lassoan** [comment]
The tricky part is spinning the extracted slice within the slice view so that the up/down and left/right directions are anatomically meaningful. Without that it is easy to get disoriented when reviewing reformatted slices.

The features that you have implemented are already awesome (I particularly like the fast and robust extraction of cross-section from models), it is perfectly fine that you don't have time to implement everything. I'll get to it in the next few days.

**Comment by chir-set** [comment]
Left/right orientation still needs to be addressed. Can be done manually, rotating 180° around the slice view's Y axis. Automatically ? Doubtful.

**Comment by chir-set** [comment]
Hi @lassoan,

Please review the latest commits when possible.

Thanks.

File: CenterlineMetrics/CenterlineMetrics.py [hunk 1]
```diff
@@ -239,20 +411,62 @@ def updateUIWithMetrics(self, value):
       self.ui.derivedDiameterValueLabel.setText(derivedDiameterStr)
     else:
       self.ui.derivedDiameterValueLabel.setText("")
-
+    # Orientation
+    self.updateSliceViewOrientationMetrics()
+    
+  def updateSliceViewOrientationMetrics(self):
+    """
```
Comment (lassoan) [inline CenterlineMetrics/CenterlineMetrics.py:418]: Please remove this comment and instead submit an issue to discuss/resolve this.

**Review by lassoan — COMMENTED** [review]
Very nice work, thank you. I've just added a trivial comment and it should be good to go.

File: CenterlineMetrics/CenterlineMetrics.py (hunk 1 above)
Comment (chir-set) [reply CenterlineMetrics/CenterlineMetrics.py]: Done. I'll open a new issue ASAP.

**Pull request merged**